packages/metro/src/DeltaBundler/traverseDependencies.js

```diff
@@ -309,12 +312,71 @@ async function addDependency<T>(
   options.onDependencyAdded();
 }
 
-function removeDependency<T>(
+/**
+ * Recursively look up `inverseDependencies` until it is empty,
+ * returning a set of paths for the last module that does not have
+ * `inverseDependencies`.
+ */
+function getAllTopLevelInverseDependencies<T>(
+  inverseDependencies: Set<string>,
+  parentModule: string,
+  graph: Graph<T>,
+  currModule: string,
+  visited: Set<string>,
+): Set<string> {
+  if (visited.has(currModule)) {
+    return new Set();
+  }
+  visited.add(currModule);
+  if (!inverseDependencies.size || currModule === parentModule) {
+    return new Set([currModule]);
+  }
+
+  return Array.from(inverseDependencies)
+    .filter(inverseDep => graph.dependencies.has(inverseDep))
+    .reduce((acc, inverseDep) => {
+      const mod = graph.dependencies.get(inverseDep);
+      if (!mod) {
```

Comment:
won't this always be false due to the `filter` above?

Reply:
I assume this is to make Flow happy, which isn't able to keep track of that information.

Reply:
Yes what @cpojer said